src/main/scala/io/citrine/lolo/trees/multitask/MultiTaskTrainingNode.scala

```diff
@@ -29,9 +32,39 @@ class MultiTaskTrainingNode(
       (Some(new MultiTaskTrainingNode(leftData, randomizePivotLocation, rng = rng)), Some(new MultiTaskTrainingNode(rightData, randomizePivotLocation, rng = rng)))
   }
 
+  // get feature importance for the i'th label
+  def getFeatureImportance(index: Int): mutable.ArraySeq[Double] = {
```

Comment:
This is similar in structure to `getNode`. I tried to combine them into a single method that creates the model node and the feature importance simultaneously, but that led to strange errors in the sparse data tests. It would somehow wind up in a situation in which the only training data on a node were missing the desired label. I could not figure out the cause, and eventually gave up and reverted to two code blocks.

Reply:
Since they are recursive calls I can see that being tricky. If you want to reduce code duplication a bit, this header block is identical:

```
    // Filter out "missing" values, which are NaN for regression and 0 for encoded categoricals
    val label = inputs.head._2(index)
    val reducedData = if (label.isInstanceOf[Double]) {
      inputs.map(x => (x._1, x._2(index).asInstanceOf[Double], x._3)).filterNot(_._2.isNaN)
    } else {
      inputs.map(x => (x._1, x._2(index).asInstanceOf[Char], x._3)).filter(_._2 > 0)
    }
    // Compute the valid data for each child
    val (left, right) = reducedData.partition(r => split.turnLeft(r._1))
```

It doesn't give you that much, but that could be branched into its own method for data filtering.

Reply:
Also, my IDE is telling me that this type cast `reducedData.asInstanceOf[Seq[(Vector[AnyVal], Double, Double)]]` is redundant, and `reducedData` has the correct type already. Maybe that's another weird scala version thing?

Reply:
> It doesn't give you that much, but that could be branched into its own method for data filtering.

Every one of those values gets used later on, so you'd have to return all of them, which makes for a hideous method.

Reply:
> Also, my IDE is telling me that this type cast reducedData.asInstanceOf[Seq[(Vector[AnyVal], Double, Double)]] is redundant, and reducedData has the correct type already. Maybe that's another weird scala version thing?

My IDE tells me that too, but if I remove the typecast then it fails to compile.

Reply:
> Every one of those values gets used later on, so you'd have to return all of them, which makes for a hideous 
 method.

Yes you are right, pardon my quick take on that suggestion.